docs/pages/tutorials/minimal/extend_schema.mdx

```diff
@@ -0,0 +1,329 @@
+
+2. Run this command in `packages/contracts` to regenerate the table libraries.
+
+   ```sh copy
+   pnpm build:mud
+   ```
+
+## Update `IncrementSystem`
+
+1. In an editor, open `packages/contracts/src/systems/IncrementSystem.sol`.
+
+   - Modify the second `import` line to import `History` (line 5)
+   - Modify the `increment` function to also update `History` (line 12).
+     To see the exact functions that are available, you can look at `packages/contracts/src/codegen/tables/History.sol` (that is the reason we ran `pnpm build:mud` to recreate it already).
+
+<CollapseCode>
+
+```solidity filename="IncrementSystem.sol.sol" copy showLineNumbers {5, 12}
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { System } from "@latticexyz/world/src/System.sol";
+import { Counter, History, HistoryData } from "../codegen/Tables.sol";
+
+contract IncrementSystem is System {
+  function increment() public returns (uint32) {
+    uint32 counter = Counter.get();
+    uint32 newValue = counter + 1;
+    Counter.set(newValue);
+    History.set(newValue, block.number, block.timestamp);
+    return newValue;
+  }
+}
+```
+
+</CollapseCode>
+
+   <details>
+   <summary>Explanation</summary>
+
+```solidity
+import { Counter, History, HistoryData } from "../codegen/Tables.sol";
+```
+
+When a table has multiple fields in the value schema, MUD generates a [Struct](https://www.tutorialspoint.com/solidity/solidity_structs.htm) to hold a full value.
+Here is `HistoryData`, copied from `packages/contract/src/codegen/History.sol`.
+
+```solidity
+struct HistoryData {
+  uint256 blockNumber;
+  uint256 time;
+}
+```
+
+Note that `IncrementSystem` doesn't need to use `HistoryData`, because it only writes to history, it doesn't read from it.
+However, this is part of manipulating the schema and therefore included in this tutorial.
+
+```solidity
+History.set(newValue, block.number, block.timestamp);
+```
+
+Set the value.
+All MUD tables have a `<table>.set` function with the parameters being the key schema fields in order and then the value schema fields in order.
+
+    </details>
+
+2. Run this command in `packages/contracts` to rebuild everything this package produces.
+
+   ```sh copy
+   pnpm build
+   ```
+
+## Update the user interface
+
+You can already run the application and see in the MUD Dev Tools that there is a `:History` table and it gets updates when you click **Increment**.
+Click the **Store data** tab and select the table **:History**.
+
+However, you can also add the history to the user interface.
+The directions here apply to the `vanilla` client template, if you use anything else you'll need to modify them as appropriate.
+
+1. Edit `packages/client/src/index.ts`.
+
+   - Import two more functions we need (lines 3-4).
+   - In `components.Counter.update$.subscribe`, add code that updates the selection of history values (lines 18-22).
+   - Add a function that is called when the user selects a different option (lines 31-37).
```

Comment:
I feel like the highlighted lines below do enough to communicate line numbers? worried about forgetting to update these when the highlighted lines are updated, or vice versa

and the collapsed explanation below walks through line by line already

Reply:
You know the audience better than I do, if you don't think we need the line numbers here I'll remove them.